common/cluster/archival_status.go

```diff
@@ -44,50 +48,53 @@ const (
 	ArchivalPaused
 	// ArchivalEnabled means this cluster is currently archiving
 	ArchivalEnabled
+
+	// DomainArchivalDisabled means a domain is not configured to handle archival
+	DomainArchivalDisabled DomainArchivalStatus = iota
+	// DomainArchivalEnabled means a domain is currently archiving
+	DomainArchivalEnabled
 )
 
 // NewArchivalConfig constructs a new valid ArchivalConfig
-func NewArchivalConfig(status ArchivalStatus, defaultBucket string, enableReadFromArchival bool) *ArchivalConfig {
+func NewArchivalConfig(
+	clusterStatus ArchivalStatus,
+	enableReadFromArchival bool,
+	domainDefaultStatus DomainArchivalStatus,
+	domainDefaultURI string,
+) *ArchivalConfig {
 	ac := &ArchivalConfig{
-		status:                 status,
-		defaultBucket:          defaultBucket,
-		enableReadFromArchival: enableReadFromArchival,
+		ClusterStatus:          clusterStatus,
+		EnableReadFromArchival: enableReadFromArchival,
+		DomainDefaultStatus:    domainDefaultStatus,
+		DomainDefaultURI:       domainDefaultURI,
 	}
-	if !ac.isValid() {
-		return &ArchivalConfig{
-			status:                 ArchivalDisabled,
-			defaultBucket:          "",
-			enableReadFromArchival: false,
-		}
+	if !ac.isDomainDefaultValid() {
```

Comment:
Can we still call this method isValid? We are checking that the whole ArchivalConfig is valid its just that the only condition that we need to actually check relates to domain fields.

Reply:
I think the method name show the fact that we are only check domain defaults and any config for the cluster level config is valid.